doc/api/api_changes.rst

```diff
@@ -37,6 +30,3 @@ added to Matplotlib, see :ref:`whats-new`
       next_api_changes/deprecations/*
       next_api_changes/development/*
       next_api_changes/removals/*
-
-.. include:: prev_api_changes/api_changes_3.4.2.rst
-.. include:: prev_api_changes/api_changes_3.4.0.rst
```

Comment:
I don't understand where 'current' API changes go on release time, if not here?

Reply:
On a release everything moves from next to the respective version section:
![grafik](https://user-images.githubusercontent.com/2836374/119615736-04847880-be00-11eb-81a0-2381c32a3541.png)

Instead of this

```
.. include:: prev_api_changes/api_changes_3.4.2.rst
```
you'll have in `release_notes.rst`
```
../api/prev_api_changes/api_changes_3.4.2.rst
```

API changes has changed it's position in the hierachry. It used to be top-levelish and contained the next changes as links to the previous changes. Now, it's only a sub topic `Release notes -> Next -> API changes`. I will rename it to `Next API Changes` to make that more clear.

Reply:
Then the `if dev` bit shouldn't be in this file, but in the new release notes file? There should be no 'next changes' for tagged releases.

This would also needs updates to the release guide.

Reply:
Yes, I've noticed the `if dev` part. I've made it structurally the same as "Next what's new" now.

I've added `if dev` around the "Release notes > Next" section. What I'm not 100% clear about is how conditional pages are handled. Due to the `if dev` we may get notes that the pages are `next_api_changes` is not in the toc tree when building the release.